ZukoAzula2016/src/org/team1540/zukoazula/Portcullis.java

```diff
@@ -21,22 +21,28 @@
     private static final FloatIO leftEncoder = leftGrabMotor.modEncoder().getEncoderPosition();
     private static final FloatIO rightEncoder = rightGrabMotor.modEncoder().getEncoderPosition();
 
-    private static final FloatInput control = ZukoAzula.controlBinding.addFloat("Portcullis Grabber Axis").deadzone(0.3f);
+    private static final FloatInput control = ZukoAzula.controlBinding.addFloat("Portcullis Grabber Axis").deadzone(0.3f).negated();
 
     private static final ArbitratedFloat leftInput = ZukoAzula.behaviors.addFloat();
     private static final ArbitratedFloat rightInput = ZukoAzula.behaviors.addFloat();
 
-    private static final FloatCell maximumSpeed = ZukoAzula.mainTuning.getFloat("Portcullis Maximum Speed", .3f);
+    private static final FloatInput pidP = ZukoAzula.mainTuning.getFloat("Portcullis PID:P", 0.001f);
+    private static final FloatInput pidI = ZukoAzula.mainTuning.getFloat("Portcullis PID:I", 0.0f);
+    private static final FloatInput pidD = ZukoAzula.mainTuning.getFloat("Portcullis PID:D", 0.0f);
 
-    public static void setup() throws ExtendedMotorFailureException {
-
-        PIDController pid = new PIDController(leftEncoder, rightEncoder.negated(), ZukoAzula.mainTuning.getFloat("Portcullis PID:P", 0.001f), ZukoAzula.mainTuning.getFloat("Portcullis PID:I", 0.0f), ZukoAzula.mainTuning.getFloat("Portcullis PID:D", 0.0f));
+    private static final FloatInput autolevelSpeed = ZukoAzula.mainTuning.getFloat("Portcullis Auto-level Speed", 0.2f);
+    private static final FloatCell maximumSpeed = ZukoAzula.mainTuning.getFloat("Portcullis Maximum Speed", 0.4f);
 
-        pid.updateWhen(FRC.constantPeriodic);
-        pid.setOutputBounds(ZukoAzula.mainTuning.getFloat("Portcullis Auto-level Speed", 0.2f));
+    public static void setup() throws ExtendedMotorFailureException {
 
-        FloatInput leftOut = control.minus(control.inRange(-0.1f, 0.1f).toFloat(0.0f, pid)).multipliedBy(maximumSpeed);
-        FloatInput rightOut = control.multipliedBy(maximumSpeed);
+        PIDController levelPID = new PIDController(leftEncoder, leftEncoder.plus(rightEncoder.negated()).dividedBy(2), pidP, pidI, pidD);
+        levelPID.updateWhen(FRC.constantPeriodic);
+        levelPID.setOutputBounds(autolevelSpeed);
+        
+        leftEncoder.eventSet(rightEncoder.negated()).on(FRC.startAuto.or(FRC.startTele));
+        
+        FloatInput leftOut = control.multipliedBy(maximumSpeed).minus(control.outsideRange(-0.1f, 0.1f).toFloat(levelPID, 0.0f));
+        FloatInput rightOut = control.multipliedBy(maximumSpeed).plus(control.outsideRange(-0.1f, 0.1f).toFloat(levelPID, 0.0f));
```

Comment:
I guess not. Or, I can just change it in my update if it is so minor. Same with the newline.